src/adcs_handler.c

```diff
@@ -46,242 +50,450 @@ ADCS_returnState adcs_telecommand(uint32_t * command, uint32_t length) {
+ * 		Success of function defined in adcs_types.h
  */
-ADCS_returnState ADCS_load_file_download_block(File_Type file_type, uint8_t counter, uint32_t offset, uint16_t block_length){
-  uint32_t command[5];
-  command[0] = LOAD_FILE_DOWNLOAD_BLOCK_ID;
-  command[1] = file_type;
-  command[2] = counter;
-  command[3] = offset;
-  command[4] = block_length;
-
-  return adcs_telecommand(command, 5);
-}
-
-/**
- * @brief
- * 		Advance File List Read Pointer
- * @return
- * 		<if applicable, brief description of what is returned by the function>
- */
-ADCS_returnState ADCS_advance_file_list_read_pointer(){
-  uint32_t command = ADVANCE_FILE_LIST_READ_POINTER_ID;
+ADCS_returnState ADCS_run_selected_program(
+    void) {  //* could be merged with the previous one in services
+  uint8_t command = RUN_SELECTED_PROGRAM_ID;
   return adcs_telecommand(&command, 1);
 }
 
 /**
  * @brief
- * 		Initiate File Upload
- * @param file_dest
- * 		Accepted parameters: EEPROM, FLASH_PROGRAM_1 - FLASH_PROGRAM_7,
- *                          SD_USER_FILE_1 - SD_USER_FILE_8
- * @param block_size
- * 
+ * 		Read the program information
+ * @param index
+ * 		Table 66 - Program index
+ * 		0 : BootLoader, 1 : Internal flash program, 2 : EEPROM,
+ * 		3-9: External flash program 1-7,
+ * 		10-17: SD user file 1-8
  * @return
- * 		<if applicable, brief description of what is returned by the function>
+ * 		Success of function defined in adcs_types.h
  */
-ADCS_returnState ADCS_initiate_file_upload(File_Upload_Dest file_dest, uint8_t block_size){
-  uint32_t command[3];
-  command[0] = INITIATE_FILE_UPLOAD_ID;
-  command[1] = file_dest;
-  command[2] = block_size;
-  adcs_telecommand(command, 3);
+ADCS_returnState ADCS_read_program_info(
+    uint8_t index) {  //* why would we need the enum
+  uint8_t command[2];
+  command[0] = READ_PROGRAM_INFO_ID;
+  if (index < 0 || index > 18) {
+    return ADCS_INVALID_PARAMETERS;
+  } else {
+    command[1] = index;
+    return adcs_telecommand(command, 2);
+  }
 }
 
 /**
  * @brief
- * 		File Upload Packet
- * @param packet_number
- * 		Packet Number
- * @param file_bytes
- * 		File Bytes
+ * 		Copy program to internal flash
+ * @param index
+ * 		Table 66 - Source Program index				//* not
+ * good for services 0 : BootLoader, 1 : Internal flash program, 2 : EEPROM,
+ * 3-9: External flash program 1-7, 10-17: SD user file 1-8
+ * @attention
+ * 		flag = 0x5A overwrites the boot segment
  * @return
- * 		<if applicable, brief description of what is returned by the function>
+ * 		Success of function defined in adcs_types.h
  */
-ADCS_returnState ADCS_file_upload_packet(uint16_t packet_number, char *file_bytes){
-  uint32_t command[3];
-  command[0] = FILE_UPLOAD_PACKET_ID;
-  command[1] = packet_number;
-  command[2] = file_bytes;
-  adcs_telecommand(command, 3);
+ADCS_returnState ADCS_copy_program_internal_flash(
+    uint8_t index, uint8_t overwrite_flag) {  //* why would we need the enum
```

Comment:
```suggestion
    uint8_t index, bool overwrite_flag) {  //* why would we need the enum
```
Set this to boolean? So for true: `command[2] = 0x5A;`

Reply:
Yeah, but it's a bit unclear in the manual. Are we sure there are no other flags? If the flag is zero, what should the `command[2]` be?

Reply:
Right, I see. Maybe it's just set to any other value (not 0x5A). But, I'm just assuming